avro2tf/src/main/scala/com/linkedin/avro2tf/jobs/Avro2TF.scala

```diff
@@ -88,6 +88,30 @@ object Avro2TF {
    */
   case class SparseVector(indices: Seq[Long], values: Seq[Float])
 
+  /**
+    * Companion object of [[SparseVector]]. It takes in extra flag filterZero. If it's set,
+    * it will filter out all zeros values and their corresponding indices
+    */
+  object SparseVector {
+    /**
+      * To construct a [[SparseVector]] based on user input
+      * @param indices indices for constructed [[SparseVector]]
+      * @param values values for constructed [[SparseVector]]. It can have zeros if filterZero is false
+      * @param filterZero it's a flag to set if filtering out zeros values and their corresponding indices
+      */
+    def apply(indices: Seq[Long], values: Seq[Float], filterZero: Boolean) : SparseVector = {
+      if (filterZero) filterOutZeroValues(indices, values) else SparseVector(indices, values)
+    }
+
+    private def filterOutZeroValues(indices: Seq[Long], values: Seq[Float]): SparseVector = {
+      val nonZerosIndices = values.zipWithIndex.map {
+        case (value, index) => (value, index)
```

Comment:
Why de-structure and re-structure the same tuple? This function is a no-op.

Reply:
you are right, will fix it. Thanks!